### PR TITLE
shortcut modal, keyboard navigation

### PR DESCRIPTION
**Pull request by JanAckermann**
## Description
Keyboard navigation:
- Dropdown keyboard navigation (up, down, enter, esc)
- Modal keyboard navigation (enter, esc) 

## Related Issue

- Part of https://github.com/owncloud/web/issues/9850

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-pkg/src/components/CreateShortcutModal.vue [hunk 1]
```diff
@@ -233,36 +366,33 @@ export default defineComponent({
...
       getParentFolderLink,
       getParentFolderName,
-      getParentFolderLinkIconAdditionalAttributes
+      getParentFolderLinkIconAdditionalAttributes,
+      onKeyEnterDrop,
+      onKeyDownDrop,
+      onKeyUpDrop,
+      onHideDrop,
+      onShowDrop,
+      onInput,
```
Comment (JanAckermann) [inline packages/web-pkg/src/components/CreateShortcutModal.vue:392]: Which Input 🤡

File: packages/web-pkg/src/components/CreateShortcutModal.vue [hunk 2]
```diff
@@ -233,36 +366,33 @@ export default defineComponent({
...
+      onClickInput,
```
Comment (JanAckermann) [inline packages/web-pkg/src/components/CreateShortcutModal.vue:393]: ^ same

File: packages/web-pkg/src/components/Search/ResourcePreview.vue [hunk 3]
```diff
@@ -81,11 +81,12 @@ export default defineComponent({
...
     const additionalAttrs = computed(() => {
       if (!props.isClickable) {
-        return {}
+        return {
+          isResourceClickable: false
+        }
       }
 
       return {
-        isResourceClickable: true,
```
Comment (JanAckermann) [inline packages/web-pkg/src/components/Search/ResourcePreview.vue:88]: default

File: packages/web-pkg/src/components/CreateShortcutModal.vue [hunk 4]
```diff
@@ -200,12 +221,124 @@ export default defineComponent({
...
+      const elements = Array.from(document.querySelectorAll('li.selectable-item'))
+      let index =
+        unref(activeDropItemIndex) !== null
+          ? unref(activeDropItemIndex)
+          : previous
+          ? elements.length
+          : -1
+      const increment = previous ? -1 : 1
+
+      do {
```
Comment (JanAckermann) [inline packages/web-pkg/src/components/CreateShortcutModal.vue:244]: I think we can kick this, irrelevant for shortcuts

**Review by lookacat — APPROVED** [review]
LGTM, nice 👍🏼

**Review by JammingBen — APPROVED** [review]

**Pull request merged**